agent/api/task/task.go

```diff
@@ -347,7 +349,7 @@ func (task *Task) ContainerByName(name string) (*apicontainer.Container, bool) {
 
 // HostVolumeByName returns the task Volume for the given a volume name in that
 // task. The second return value indicates the presence of that volume
-func (task *Task) HostVolumeByName(name string) (HostVolume, bool) {
+func (task *Task) HostVolumeByName(name string) (taskresourcevolume.HostVolume, bool) {
```

Comment:
Is the hostvolume something the user is volunteering to be used by the task? Or are we making that assumption for the user?

Concern:
It seems risky to make any assumptions related to disk space usage since there are limited tools, telemetry, and "rails" to manage disk utilization on the host to prevent disk space related issues.

Reply:
from moby issue #21829 (thought it might be relevant):
What is the volume size of the default volume?

Volumes created through the default ("local") volume driver don't have a size limit; they're basically just directories created inside /var/lib/docker/volumes. If you want to put a size constraint on volumes, you need to use a volume-plugin that supports this. Some plugins can be found in the documentation.